src/components/inspector/panels/Panels.tsx

```diff
@@ -54,19 +53,53 @@ import TabPanel from './components/TabPanel'
 import StatArrowPanel from './components/StatArrowPanel'
 import StatLabelPanel from './components/StatLabelPanel'
 import SkeletonPanel from './components/SkeletonPanel'
-import CCPanel from '~custom-components/inspector/panels/components/CCPanel'
-import SamplePanel from '~custom-components/inspector/panels/components/SamplePanel'
+import { useSelector } from 'react-redux'
+import { getCustomComponentNames } from '~core/selectors/customComponents'
+import { onboarding } from '~templates/onboarding'
+import { convertToPascal } from '~components/editor/Editor'
+
+const importView = (component: any) => {
+  component = convertToPascal(component)
+  return lazy(() =>
+    import(
+      `src/custom-components/inspector/panels/components/${component}Panel.oc.tsx`
+    ).catch(() => import('src/custom-components/fallback')),
+  )
+}
 
-const Panels: React.FC<{ component: IComponent; isRoot: boolean }> = ({
-  component,
-  isRoot,
-}) => {
+const Panels: React.FC<{
+  component: IComponent
+  isRoot: boolean
+  isCustom: boolean
+}> = ({ component, isRoot, isCustom }) => {
   const { type } = component
+  const [views, setViews] = useState<any>([])
+  const customComponents = useSelector(getCustomComponentNames)
+
+  useEffect(() => {
+    async function loadViews() {
+      const componentPromises = await customComponents.map(
+        async (component: any) => {
+          const View = await importView(component)
+          return <View key={component} />
+        },
+      )
+      Promise.all(componentPromises).then(setViews)
+    }
+    loadViews()
+  }, [customComponents])
```

Comment:
Just the component names or component jsons?

Reply:
just the names